libcxx/include/__utility/pair.h

```diff
@@ -633,42 +633,42 @@ get(const pair<_T1, _T2>&& __p) _NOEXCEPT {
 #if _LIBCPP_STD_VER >= 14
 template <class _T1, class _T2>
 inline _LIBCPP_HIDE_FROM_ABI constexpr _T1& get(pair<_T1, _T2>& __p) _NOEXCEPT {
-  return __get_pair<0>::get(__p);
+  return __p.first;
 }
 
 template <class _T1, class _T2>
 inline _LIBCPP_HIDE_FROM_ABI constexpr _T1 const& get(pair<_T1, _T2> const& __p) _NOEXCEPT {
-  return __get_pair<0>::get(__p);
+  return __p.first;
 }
 
 template <class _T1, class _T2>
 inline _LIBCPP_HIDE_FROM_ABI constexpr _T1&& get(pair<_T1, _T2>&& __p) _NOEXCEPT {
-  return __get_pair<0>::get(std::move(__p));
+  return std::forward<_T1&&>(__p.first);
```

Comment:
Don't we mean `std::move(__p.first)`?

Reply:
No. If `_T1` is an lvalue reference we want it to stay an lvalue reference, even if we get the `pair` as an rvalue. (I got that wrong on my first try of this patch as well - maybe a comment would be good)

Reply:
Ah, you're right. But I think `std::forward<_T1>(__p.first)` is equivalent (due to reference collapsing rules), and is what we generally use elsewhere. Any reason for using `std::forward<_T1&&>`?

Reply:
I don't think it is equivalent if `_T1` is a prvalue.

Reply:
I think they are equivalent due to reference collapsing rules. `T&& &&` is the same as `T&&`. I'm not attached to the spelling, though.